orderer/consensus/smartbft/synchronizer_bft.go

```diff
@@ -0,0 +1,239 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package smartbft
+
+import (
+	"sort"
+	"time"
+
+	"github.com/SmartBFT-Go/consensus/pkg/types"
+	"github.com/SmartBFT-Go/consensus/smartbftprotos"
+	"github.com/hyperledger/fabric-lib-go/bccsp"
+	"github.com/hyperledger/fabric-lib-go/common/flogging"
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric/common/deliverclient"
+	"github.com/hyperledger/fabric/internal/pkg/peer/blocksprovider"
+	"github.com/hyperledger/fabric/internal/pkg/peer/orderers"
+	"github.com/hyperledger/fabric/orderer/common/cluster"
+	"github.com/hyperledger/fabric/orderer/common/localconfig"
+	"github.com/hyperledger/fabric/orderer/consensus"
+	"github.com/hyperledger/fabric/protoutil"
+	"github.com/pkg/errors"
+)
+
+type BFTSynchronizer struct {
+	lastReconfig       types.Reconfig
+	selfID             uint64
+	LatestConfig       func() (types.Configuration, []uint64)
+	BlockToDecision    func(*common.Block) *types.Decision
+	OnCommit           func(*common.Block) types.Reconfig
+	Support            consensus.ConsenterSupport
+	CryptoProvider     bccsp.BCCSP
+	BlockPuller        BlockPuller              // TODO improve - this only an endpoint prober - detect self EP
+	clusterDialer      *cluster.PredicateDialer // TODO make bft-synchro
+	localConfigCluster localconfig.Cluster      // TODO make bft-synchro
+	Logger             *flogging.FabricLogger
+}
+
+func (s *BFTSynchronizer) Sync() types.SyncResponse {
+	decision, err := s.synchronize()
+	if err != nil {
+		s.Logger.Warnf("Could not synchronize with remote orderers due to %s, returning state from local ledger", err)
+		block := s.Support.Block(s.Support.Height() - 1)
+		config, nodes := s.LatestConfig()
+		return types.SyncResponse{
+			Latest: *s.BlockToDecision(block),
+			Reconfig: types.ReconfigSync{
+				InReplicatedDecisions: false, // If we read from ledger we do not need to reconfigure.
+				CurrentNodes:          nodes,
+				CurrentConfig:         config,
+			},
+		}
+	}
+
+	// After sync has ended, reset the state of the last reconfig.
+	defer func() {
+		s.lastReconfig = types.Reconfig{}
+	}()
+	return types.SyncResponse{
+		Latest: *decision,
+		Reconfig: types.ReconfigSync{
+			InReplicatedDecisions: s.lastReconfig.InLatestDecision,
+			CurrentConfig:         s.lastReconfig.CurrentConfig,
+			CurrentNodes:          s.lastReconfig.CurrentNodes,
+		},
+	}
+}
+
+func (s *BFTSynchronizer) synchronize() (*types.Decision, error) {
+	//=== We use the BlockPuller to probe all the endpoints and establish a target height, as well as detect
+	// the self endpoint.
+
+	// In BFT it is highly recommended that the channel/orderer-endpoints (for delivery & broadcast) map 1:1 to the
```

Comment:
while that is true, i don't see how it helps us having this comment written here, since the people that develop Fabric don't use it :-) 

Maybe instead move it to `configtx.yaml` ?

Reply:
Good suggestion.